### PR TITLE
Adds an option to skip downloading binaries

### PR DESCRIPTION
**Pull request by eaviles**
We are using your module in our company and since we are behind a corporate proxy installs would stall. Please consider this PR that adds a `DD_SKIP_DOWNLOAD` environment variable. If `true`, the `post_install` script will try to rebuild the binaries without attempting to download the pre-built binaries from GitHub.

Let me know if you need me to do any changes and thanks in advance.

### CODE REVIEW

**Comment by rochdev** [comment]
Do you know what causes the script to hang? While I think the ability to opt out of the download is a great idea, I would also like the script to not hang in the first place. The fallback should occur quickly and be transparent. Then opting out would be used only if there is for example some kind of incompatibility with the environment.

File: scripts/post_install.js [hunk 1]
```diff
@@ -13,12 +13,16 @@ const pkg = require('../package.json')
 const name = `${os.platform()}-${os.arch()}`
 
 if (process.env.DD_NATIVE_METRICS !== 'false') {
-  download(`v${pkg.version}`)
-    .catch(() => getLatestTag().then(download))
-    .then(persist)
-    .then(extract)
-    .then(cleanup)
-    .catch(rebuild)
+  if (process.env.DD_SKIP_DOWNLOAD !== 'true') {
```
Comment (rochdev) [inline scripts/post_install.js]: I would prefer a less generic name. Maybe something like `DD_SKIP_PREBUILT_DOWNLOAD`?

**Comment by eaviles** [comment]
> Do you know what causes the script to hang? While I think the ability to opt out of the download is a great idea, I would also like the script to not hang in the first place. The fallback should occur quickly and be transparent. Then opting out would be used only if there is for example some kind of incompatibility with the environment.

@rochdev we had the same issue with other tools too when trying to make a request outside our network without the proxy credentials. Our enterprise proxy would just hang. We have tried figuring it out but haven't been lucky yet. In general we are not allowed to use tarballs, zips, or anything other than the files coming from out private NPM registry. We do use the `--build-from-source` argument to skip binaries in other modules.

File: scripts/post_install.js (hunk 1 above)
Comment (eaviles) [reply scripts/post_install.js]: 👍 :

**Comment by rochdev** [comment]
I think the original issue is because the proxy drops the TCP packet required to know if the connection was established or not, which results in the socket waiting indefinitely for a response. I added a timeout to the script to prevent this from happening in #592.

However, for your case where you know in advance that you cannot download anything, then an environment variable definitely makes more sense to prevent the download in the first place.

Thanks a lot for the contribution!

**Comment by rochdev** [comment]
@eaviles Just a heads up that we ended up adding support for `--build-from-source` and the corresponding environment variable in #594, which will replace the environment variable from this PR when we release the next version.

**Pull request merged**